Lib/test/libregrtest/runtest_mp.py

```diff
@@ -149,16 +156,21 @@ def _kill(self):
             return
         self._killed = True
 
-        print(f"Kill {self}", file=sys.stderr, flush=True)
+        if USE_PROCESS_GROUP:
+            what = f"{self} process group"
+        else:
+            what = f"{self}"
+
+        print(f"Kill {what}", file=sys.stderr, flush=True)
         try:
-            popen.kill()
+            if USE_PROCESS_GROUP:
+                os.killpg(popen.pid, signal.SIGKILL)
```

Comment:
Shouldn't you call `popen.wait()` after this?

Reply:
fwiw, the old/existing code didn't.  (`popen.kill()` does not call `.wait()`)

Reply:
popen.wait() is always called.

The main thread calls stop() which calls _kill().

wait() is called in the TestWorkerProcess thread: _run_process() ends with "finally: self._wait_completed()" to ensure that we also read the exit status to prevent leaking a zombie process.

In my experience, calling wait() from 2 different threads is causing too many troubles :-)